### PR TITLE
chore: dl default model from huggingface

### PR DESCRIPTION
**Pull request by qued**
We've now seen cases where dropbox links are blacklisted, which prevent them from downloading the default detectron2 model. This PR changes the default model retrieval method to use huggingface (which already hosts the weights and config for the model). The code is also updated so getting the default model and a non-default model uses more of the same code.

 #### Testing:

Check that the `$HOME/.cache/huggingface/hub/models--layoutparser--detectron2/` folder does _not_ exist (if it does, delete it).

Run:
```python
from unstructured_inference.models import get_model
get_model()
```
This should result in the model being downloaded from hf hub. Check that the folder mentioned above has been created and contains 2 files, one a binary of size ~330MB, the other a plaintext YAML of 5.37kB.

### CODE REVIEW

**Comment by qued** [comment]
Going back to draft mode while I refactor to try to get stuff out of `models.__init__.py`.

**Review by MthwRobinson — APPROVED** [review]
Couple questions on naming that I'd consider optional. Otherwise, LGTM!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
-## 0.2.4-dev1
+## 0.2.4-dev2
```
[the remark this answers is not in the document]
Comment (qued) [reply CHANGELOG.md]: Bumped in [9e19f4b](https://github.com/Unstructured-IO/unstructured-inference/pull/30/commits/9e19f4b955ce46ead0a4a158b30bbf225454a815).

File: unstructured_inference/models/base.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+from unstructured_inference.models.detectron2 import (
+    load_model,
+    Detectron2LayoutModel,
+    DEFAULT_LABEL_MAP,
+    DEFAULT_EXTRA_CONFIG,
+)
+
+
+def get_model(model: Optional[str] = None) -> Detectron2LayoutModel:
```
[the remark this answers is not in the document]
Comment (qued) [reply unstructured_inference/models/base.py:12]: Per our Slack conversation, I think we should move to some sort of generic model wrapper class soon as the output here, but that's probably out of scope for this PR. So leaving it for now pending changes coming soon.

**Pull request merged**